### ISSUE TITLE
[Core] Minor: measure text function taking a pointer is redundant

### ISSUE THREAD

**Issue by mikejsavage** [nicbarker/clay#74]
Clay's text measuring callback is defined to take a pointer to `Clay_String`, which seems redundant because Clay_String is already a pointer.

I may be wrong but it looks like the rest of the codebase takes Clay_String directly, besides some things that are downstream of MeasureText.

**Comment by nicbarker** [comment]
Yes, I'd say you're probably right. It will be a breaking change at this point though, so I'll have to do it soon if it were to change 😅

**Comment by Muzz** [comment]
Imo it's a new library, you should just make breaking changes for the first few versions, and focus on nailing all this stuff in the short term.

**Comment by nicbarker** [comment]
This was fixed here https://github.com/nicbarker/clay/commit/326325ffaf65770c2c32357510a132bea2f5d3b3 🙂
